### PR TITLE
feat(edgeless): support adding edgeless templates

### PR DESCRIPTION
**Pull request by doouding**

### CODE REVIEW

**Comment by doouding** [comment]
The PR is ready for review, but don't merge it yet.

File: packages/blocks/src/surface-block/service/template-middlewares.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import { assertType } from '@blocksuite/global/utils';
+import { type SnapshotReturn } from '@blocksuite/store';
+
+import { generateElementId } from '../index.js';
+import type { SlotBlockPayload, TemplateJob } from './template.js';
+
+export const replaceIdMiddleware = (job: TemplateJob) => {
```
Comment (Saul-Mirone) [inline packages/blocks/src/surface-block/service/template-middlewares.ts:7]: naming

File: packages/store/src/transformer/job.ts [hunk 2]
```diff
@@ -213,6 +213,25 @@ export class Job {
     return model;
   }
 
+  async snapshotToModelData(snapshot: BlockSnapshot) {
```
Comment (Saul-Mirone) [inline packages/store/src/transformer/job.ts:216]: Please keep using the arrow method like other methods.

Comment (doodlewind) [reply packages/store/src/transformer/job.ts:216]: Most places use default method by default. But aligning to the nearby implementation surely makes more sense.

File: packages/blocks/src/page-block/edgeless/components/toolbar/template/builtin-templates.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  asserts?: Record<string, string>;
+  preview: string;
+};
+
+export type TemplateCategory = {
+  name: string;
+  templates: Template[];
+};
+
+interface ITemplateManager {
```
Comment (doodlewind) [inline packages/blocks/src/page-block/edgeless/components/toolbar/template/builtin-templates.ts:15]: I think we are deprecating the `ISomething` naming convention.

**Review by doodlewind — APPROVED** [review]
Thanks for your hard work!

**Pull request merged**